src/Resources/skeleton/crud/test/Test.tpl.php

```diff
@@ -0,0 +1,134 @@
+<?= "<?php\n" ?>
+
+namespace <?= $namespace ?>;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+
+class <?= $class_name ?> extends WebTestCase<?= "\n" ?>
+{
+    public function testIndex()
+    {
+        $client = static::createClient();
+        $crawler = $client->request('GET', '<?= $route_path ?>/');
+
+        $this->assertSame(200, $client->getResponse()->getStatusCode());
+        $this->assertContains('<!DOCTYPE html>', $client->getResponse()->getContent());
+        $this->assertContains('<?= $entity_class_name ?> index', $crawler->filter('h1')->text());
+
+        $newLink = $crawler->filter('a:contains("Create new")')->eq(0)->link();
+
+        return $newLink;
+    }
+
+    /**
+     * @TODO implement new test
+     * @depends testIndex
+     */
+    public function testNew($newLink)
+    {
+        $client = static::createClient();
+        $crawler = $client->click($newLink);
+
+        $this->assertSame(200, $client->getResponse()->getStatusCode());
+        $this->assertContains('<!DOCTYPE html>', $client->getResponse()->getContent());
+        $this->assertContains('Create new <?= $entity_class_name ?>', $crawler->filter('h1')->text());
+
+        $form = $crawler->selectButton('Save')->form();
+
+        // @TODO Set valid values for your fields
+<?php foreach ($form_fields as $form_field): ?>
+        // $form['<?= $route_name ?>[<?= $form_field ?>]']->setValue('Lorem ipsum');
+<?php endforeach; ?>
+        // $client->submit($form);
+
+        // $this->assertTrue($client->getResponse()->isRedirect());
+
+        // $crawler = $client->followRedirect();
+        // $this->assertTrue($client->getResponse()->isSuccessful());
+        // $this->assertContains('<!DOCTYPE html>', $client->getResponse()->getContent());
+        // $this->assertContains('<?= $entity_class_name ?> index', $crawler->filter('h1')->text());
+        // $this->assertContains('<td>Lorem ipsum</td>', $client->getResponse()->getContent());
+    }
+
+    // /**
+    //  * @TODO implement edit test
+    //  * @depends testNew
```

Comment:
Instead of depending on testNew, we should create an entity directly via the EntityManager here and persist it. Depending on `testNew` is convenient at first, but it couples the test methods together, which has a practical downside: I now need to run two full tests before running this test (testIndex, then testNew, then this one).

Btw, I'm not so sure that we need to comment out *all* of the edit tests. Because... we *know* what entity is being generated. And so, we could write some code to instantiate that object and persist it. Heck... thinking crazy, we could use faker and introspect the Doctrine annotation information to even fill in the required fields if they have Faker installed (we could recommend it).

It's a bit more work, but in theory, we could make these tests fully (or nearly fully) functional.

Reply:
I have looked into using Faker for the test generation.

What is easily achievable is using Faker inside the tests to populate Entities based on their properties. 
However, this will lead to tests never failing when properties were changed. Faker will always use an up-to-date generator, because the guesser bases on the current entity 
Tests will only fail if the `Form` is not updated and requires a change, too.

This looks like:
```php
// App\Test\Controller\ExampleControllerTest;
// inside a test* method that needs an entity in the db:
$entityPopulator = new EntityPopulator($this->entityManager->getClassMetadata(Test::class));
$entityPopulator->setColumnFormatters($entityPopulator->guessColumnFormatters($this->fakerGenerator));

/** @var App\Entity\Test */
$test = $entityPopulator->execute($this->entityManager, []);

$this->entityManager->persist($test);
$this->entityManager->flush();
```

One could argument that this is enough, because we are testing the controller - not the entity or form.
I'd prefer to generate tests that contain somewhat fixed values and will fail, when either the entity or form are changed.
So tests could either contain faker generator calls ``$faker->name()`` or the output of this. The former would require reverse-engineering Faker's name guesser, as the populators are returned as closures and do not allow retrieving their name.
This means we must keep a copy of this up to date, to allow getting the generator names as string: https://github.com/fzaninotto/Faker/blob/master/src/Faker/Guesser/Name.php
Not the best option.

This is also complicated, due to association mappings.
We can leave them out or auto-populate them in the tests.
If we leave them out, the generated tests will fail - our test of the maker command must reflect this.

WDYT?